WINDUPRULE-898 EAP8 RESTEasy rules

PhilipCattanach opened the pull request:
Hi Mark,
When you get a moment can you review this ruleset please? 
Please have a look at the document attached to the Jira just to be sure that I've covered all of the issues.
No rush.
Thanks,
Phil

mrizzi commented on rules/rules-reviewed/eap8/eap6/eap8-resteasy.windup.xml:
```diff
@@ -0,0 +1,172 @@
+            </when>
+            <perform>
+                <hint title="The org.jboss.resteasy resteasy-jettison-provider artifact has been removed" effort="5" category-id="mandatory">
+                    <message>The `org.jboss.resteasy` `resteasy-jettison-provider` artifact has been removed. There is no documented solution for this issue.</message>
```
```suggestion
                    <message>The `org.jboss.resteasy:resteasy-jettison-provider` artifact has been removed. There is no documented solution for this issue.</message>
```

PhilipCattanach replied:
That will require a change to the test as well

PhilipCattanach commented:
Hi @m-brophy 
I have updated the ruleset so the issues that have no documented solution are 13 SP and changed formatting of the group and artifacts referenced so they are in the style of `group:artifact`. When the automated tests pass can you review the changes please?
